File: js/examples/python/html_download.py
import sys
import requests
import os

url = sys.argv[1] if len(sys.argv) > 1 else 'https://example.com'
endpoint = f'http://localhost:3000/html?url={url}'

response = requests.get(endpoint)
response.raise_for_status()

output_path = os.path.join(os.path.dirname(__file__), 'downloaded.html')
with open(output_path, 'w', encoding='utf-8') as f:
    f.write(response.text)

print(f'HTML saved to {output_path}') 

File: js/examples/js/html_download.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const url = process.argv[2] || 'https://example.com';
const endpoint = `http://localhost:3000/html?url=${encodeURIComponent(url)}`;

fetch(endpoint)
  .then((res) => {
    if (!res.ok) {
      throw new Error(`HTTP error! status: ${res.status}`);
    }
    return res.text();
  })
  .then((text) => {
    const outputPath = path.join(__dirname, 'downloaded.html');
    fs.writeFileSync(outputPath, text, 'utf-8');
    console.log(`HTML saved to ${outputPath}`);
  })
  .catch((err) => {
    console.error('Error:', err);
  });
